packages/blocks/src/page-block/default/utils.ts

```diff
@@ -127,29 +128,67 @@ export async function downloadImage(model: BaseBlockModel) {
   };
 }
 
-export async function copyImgToClip(model: BaseBlockModel) {
-  const url = await getUrlByModel(model);
-  assertExists(url);
+export async function copyImgToClip(model: EmbedBlockModel) {
+  const copyType = 'blocksuite/x-c+w';
+  const text = model.block2Text('', 0, 0);
+  const delta = [
+    {
+      insert: text,
+    },
+  ];
+  const copyData = JSON.stringify({
+    data: [
+      {
+        type: model.type,
+        sourceId: model.sourceId,
+        width: model.width,
+        height: model.height,
+        caption: model.caption,
+        flavour: model.flavour,
+        text: delta,
+        children: model.children,
+      },
+    ],
+  });
+  const copySuccess = _copyToClipboardFromPc([
+    { mimeType: copyType, data: copyData },
+  ]);
+  copySuccess && toast('Copied image to clipboard');
+}
+interface clipItem {
+  mimeType: string;
+  data: string;
+}
 
-  const newImage = new Image();
-  newImage.src = url;
-  newImage.onload = () => {
-    const canvas = document.createElement('canvas');
```

Comment:
I thought the `downloadImage` function in this module should also be refactored.

Reply:
But this pr is about downloading, and I think we can open a new branch to solve this problem.